sdk/lib/async/stream.dart

```diff
@@ -1776,11 +1778,14 @@ abstract class StreamIterator<T> {
    * Returns a future which will complete with either `true` or `false`.
    * Completing with `true` means that another event has been received and
    * can be read as [current].
-   * Completing with `false` means that the stream itearation is done and
+   * Completing with `false` means that the stream iteration is done and
    * no further events will ever be available.
    * The future may complete with an error, if the stream produces an error,
    * which also ends iteration.
    *
+   * The stream is paused when the future completes with `true` and
+   * resumed on the next call to `moveNext`.
+   *
```

Comment:
This is almost true - if you call `moveNext` immediately in the  callback of the previous `moveNext` calls's future, then it doesn't pause at all.
Otherwise it's correct. The current behavior is a recent change, previous to that, the iterator had a one-element cache, and it didn't pause as long as you asked for the next `moveNext` before the event occurred. Now it pauses unless you call `moveNext` immediately - but that happens in many cases anyway, and the code is much simpler.